### PR TITLE
fix(cli): 'cdk synth' not able to fail if stacks have errors

### LINKED ISSUE #9226: [pipelines] Metadata errors should fail synthesis (on stacks deployed in the pipeline)
This issue is similar to https://github.com/aws/aws-cdk/issues/5594 however, occurs without nested stacks.
Instead, I have a cross-account CdkPipeline setup according to the minimal example from the documentation.
My Stack consists only of a simple VPC.

When deploying, the CloudFormation template in the target account still contains the dummy1a, dummy1b etc. AZs.
Finally, these are reported as invalid and my stack is rolled back.

### Reproduction Steps

```
pipeline.addApplicationStage(new MyApplication(this, 'Dev', {
      env: {
        account: '11111111111',
        region: 'eu-west-1',
      }
    }));
```
Where MyApplication consist of a single stack with only `new VPC(this, 'VPCName')`

### Error Log

```
Value (dummy1a) for parameter availabilityZone is invalid. Subnets can currently only be created in the following availability zones: eu-west-1a, eu-west-1b, eu-west-1c. (Service: AmazonEC2; Status Code: 400; Error Code: InvalidParameterValue
```

### Environment

  - **CLI Version      :** 1.54.0
  - **Framework Version:** 1.54.0
  - **Node.js Version:**  14.4.0
  - **OS               :**
  - **Language (Version):**  TypeScript (3.8.3)

### Other

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by otaviomacedo**
The environment variable `STACKS_TO_VALIDATE` contains the stack hierarchical ids that should be validated after synthesis. By default (when the variable is undefined), it validates only the printed stacks, keeping backward compatibility.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk/lib/cdk-toolkit.ts [hunk 1]
```diff
@@ -394,14 +394,15 @@ export class CdkToolkit {
 
   private async selectStacksForDiff(stackNames: string[], exclusively?: boolean) {
```
Comment (rix0rrr) [inline packages/aws-cdk/lib/cdk-toolkit.ts:395]: `selectStacksForDiff` ? Is that just a poorly chosen name?

It's a bit confusing in this context.

File: packages/aws-cdk/lib/cdk-toolkit.ts [hunk 2]
```diff
@@ -394,14 +394,15 @@ export class CdkToolkit {
...
     const assembly = await this.assembly();
-    const stacks = await assembly.selectStacks(stackNames, {
+
+    const idsToValidate = process.env.STACKS_TO_VALIDATE ? process.env.STACKS_TO_VALIDATE.split(';') : stackNames;
```
Comment (rix0rrr) [inline packages/aws-cdk/lib/cdk-toolkit.ts:398]: Did we confirm that the method of using an environment variable is going to scale to large sizes? Or are we going to use wildcards?

File: packages/aws-cdk/lib/cdk-toolkit.ts (hunk 1 above)
Comment (otaviomacedo) [reply packages/aws-cdk/lib/cdk-toolkit.ts:395]: The same stacks are selected for diff and synthesis. This method is reused in both contexts.

File: packages/aws-cdk/lib/cdk-toolkit.ts (hunk 2 above)
Comment (otaviomacedo) [reply packages/aws-cdk/lib/cdk-toolkit.ts:398]: I've tried with values as long as 50K characters and it worked fine.

**Review by rix0rrr — APPROVED** [review]

**Pull request merged**